frontend/src/components/TE/TimestampGiven.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+  <div>
+    {{ timestamp }}
+    <p class="subtext" v-if="daysPassed">{{ daysPassed }}</p>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Mixins } from "vue-property-decorator";
+import moment from "moment";
+import FormatTime from "@/mixins/FormatTime";
+
+@Component({
+  name: "TimestampGiven"
+})
+export default class DateGiven extends Mixins<FormatTime>(FormatTime) {
```

Comment:
This mixin can be useful outside TE section in other sections, so make it generic (common).
And why the class name differs from the component name?